core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -109,6 +123,13 @@ private[spark] class CoarseMesosSchedulerBackend(
     }
   }
 
+  protected def driverUrl: String = AkkaUtils.address(
+    AkkaUtils.protocol(sc.env.actorSystem),
+    SparkEnv.driverActorSystemName,
+    conf.get("spark.driver.host"),
+    conf.get("spark.driver.port"),
+    CoarseGrainedSchedulerBackend.ACTOR_NAME)
```

Comment:
Just a small correction, the value of that setting seems to be `1`, so `toBoolean` fails. You can just test for `conf.exists("spark.testing")`.